EntityFramework.Utilities/EntityFramework.Utilities/EFBatchOperation.cs

```diff
@@ -232,6 +239,16 @@ public int Update<TP>(Expression<Func<T, TP>> prop, Expression<Func<T, TP>> modi
 			var mquery = (ObjectQuery<T>)_context.CreateObjectSet<T>().Where(updateExpression);
 			var mqueryInfo = provider.GetQueryInformation(mquery);
 
+			// If Entity Framework has optimized either query down to a query that queries against nothing and returns
+			// nothing, then abort as there is no way to somehow translate it to a delete query.
```

Comment:
@RudeySH Random thing I just noticed, if you want to fix it when you merge this into the refactors, this comment says "delete" when it should say "update," apologies.